tests/suite/math/call.typ

```diff
@@ -8,6 +8,97 @@ $ pi(a,) $
 $ pi(a,b) $
 $ pi(a,b,) $
 
+--- math-call-unclosed-func ---
+#let func(x) = x
+// Error: 1-2 unclosed delimiter
+// Error: 6-7 unclosed delimiter
+// Error: 8-9 unexpected dollar sign
+$func(a$
+
+--- math-call-unclosed-non-func ---
+// Error: 1-2 unclosed delimiter
+// Error: 5-6 unclosed delimiter
+// Error: 7-8 unexpected dollar sign
+$sin(x$
+
+--- math-call-named-args ---
+#let func1(my: none) = my
+#let func2(_my: none) = _my
+#let func3(my-body: none) = my-body
+#let func4(_my-body: none) = _my-body
+#let func5(m: none) = m
+$ func1(my: a) $
+$ func2(_my: a) $
+$ func3(my-body: a) $
+$ func4(_my-body: a) $
+$ func5(m: a) $
+$ func5(m: sigma : f) $
+$ func5(m: sigma:pi) $
+
+--- math-call-named-args-duplicate ---
+#let func(my: none) = my
+// Error: 15-17 duplicate argument: my
+$ func(my: a, my: b) $
+
+--- math-call-named-single-underscore ---
+#let func(x) = x
+// Error: 8-9 expected identifier, got underscore
+$ func(_: a) $
+
+--- math-call-named-single-char-error ---
+#let func(m: none) = m
+// Error: 8-13 unexpected argument
+$ func(m : a) $
+
+--- math-call-named-args-repr ---
+#let args(..body) = body
+#let check(it, r) = test-repr(it.body.text, r)
+#check($args(_a: a)$, "(_a: [a])")
+#check($args(_a-b: a)$, "(_a-b: [a])")
+#check($args(a-b: a)$, "(a-b: [a])")
+#check($args(a-b-c: a)$, "(a-b-c: [a])")
+#check($args(a--c: a)$, "(a--c: [a])")
+#check($args(a: a-b)$, "(a: sequence([a], [−], [b]))")
+#check($args(a-b: a-b)$, "(a-b: sequence([a], [−], [b]))")
+#check($args(a-b)$, "(sequence([a], [−], [b]))")
+
+--- math-call-spread-content-error ---
+#let args(..body) = body
+// Error: 7-16 cannot spread content
+$args(..(a + b))$
+
+--- math-call-spread-multiple-exprs ---
+#let args(..body) = body
+// Error: 11-12 unexpected text
```

Comment:
the error message is not really ideal here. not sure whether we can improve, but in principle a missing comma error would be nicer.

Reply:
I think we can just change this line to improve it.
https://github.com/typst/typst/blob/b44c90ef33ec1789bdd9a5c8d614e9ef2f43e05c/crates/typst-syntax/src/parser.rs#L512
Though you could also put a semicolon instead of a comma.

Reply:
What if instead of parsing one math expression after dots, we parse until the next comma? Then the error here would happen during evaluation.

Changing `math_expr(p)` to `math_exprs(p, syntax_set!(...))` when parsing the spread, the error becomes: "Error: 7-14 (`..a + b`) cannot spread content"

Reply:
Yeah I'm not sure what I was saying before, definitely can't change what I suggested.

I changed it at some point to only parse one expression after dots. I'll change it back, I think that makes this error a lot nicer.

Reply:
Actually, never mind, I think what I was saying before makes sense. I just changed that line to `p.expected("comma or semicolon")`

```
error: expected comma or semicolon
  ┌─ tests/suite/playground.typ:3:9
  │
3 │ $args(..a + b)$
  │          ^
```

Reply:
See, I'm not sure that it really is the error message we want, because it doesn't help the user understand *why*.

Reply:
There should be at least some indication that the parser tried to parse a spread expression here.

Reply:
I'd argue that this already kind of does that. It's saying a comma or semicolon was expected after the `a`, signalling to a user that the dots only apply to a single expression.

Reply:
Yes, but after they insert a comma they'll get one more error: "cannot spread content"

Reply:
Hmm. The best error is context-sensitive. If the item after the dots is spreadable, then they're missing a comma. If the item after the dots isn't, then they need to escape the dots or something

Reply:
That makes sense though I'd think, so I don't see the issue with getting the other error. The `a` is content after all.